app/src/main/kotlin/com/exallium/todoapp/notedetail/NoteDetailPresenter.kt

```diff
@@ -0,0 +1,38 @@
+package com.exallium.todoapp.notedetail
+
+import com.exallium.todoapp.entities.Note
+import com.exallium.todoapp.mvp.BasePresenter
+import rx.Single
+import rx.SingleSubscriber
+import timber.log.Timber
+
+/**
+ * Presenter for Note Detail Screen
+ */
+class NoteDetailPresenter(val view: NoteDetailView, val model: NoteDetailModel) : BasePresenter<NoteDetailView>(view) {
+    lateinit var note: Note
```

Comment:
This is unnecessary I think.  You can always get the current state of the note from the model, so you shouldn't need to keep a copy of this yourself.

When the user saves a note for example, you can do:

```
view.saveClicks()
    .flatMap { model.getNoteById(noteId) }
    .map { it.with(/* ... fields ... */)
    .subscribe { model.saveNote(it) }
```